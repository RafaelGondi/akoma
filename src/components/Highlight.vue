<!-- eslint-disable vue/multi-word-component-names -->
<template>
	<span
		class="highlight__container"
		:class="dynamicHighlightClass"
		:style="dynamicStyle"
	>
		<!-- @slot Slot usado para especificar o texto que receberá o highlight. -->
		<slot />
	</span>
</template>

<script>
export default {
	props: {
		/**
		 * A variante do Highlight. São 3 variantes implementadas: 'info', 'success'
		 * e 'danger'
		 */
		variant: {
			type: String,
			default: 'info',
		},
		/**
		 * Especifica se o highlight vai ser estático ou animado.
		 */
		animated: {
			type: Boolean,
			default: false,
		},
		/**
		 * Especifica a duração da animação.
		 */
		duration: {
			type: Number,
			default: 1,
		},
		/**
		 * O tempo de espera até a animação começar.
		 */
		delay: {
			type: Number,
			default: 0,
		},
	},

	computed: {
		dynamicHighlightClass() {
			let dynamicClass = '';

			if (this.animated) {
				dynamicClass = 'highlight__container--highlighted';
			}

			switch (this.variant) {
				case 'info':
					return `${dynamicClass} highlight__container--info`;
				case 'success':
					return `${dynamicClass} highlight__container--success`;
				case 'danger':
					return `${dynamicClass} highlight__container--danger`;
				default:
					return `${dynamicClass} highlight__container--info`;
			}
		},

		dynamicStyle() {
			return {
				'--duration': `${this.duration}s`,
				'--delay': `${this.delay}s`,
			};
		},
	},
};
</script>
<style lang="scss" scoped>
@import '../assets/sass/tokens.scss';

.highlight__container {
	background-size: 200%;
	border-radius: 4px;
	padding: pa(1);

	&--highlighted {
		animation: highlight var(--duration) ease-in var(--delay) backwards;
	}

	&--info {
		background-image: linear-gradient(to right, rgba($bn-300, .25) 50%, transparent 50%);
	}

	&--success {
		background-image: linear-gradient(to right, rgba($gp-300, .28) 50%, transparent 50%);
	}

	&--danger {
		background-image: linear-gradient(to right, rgba($rc-300, .22) 50%, transparent 50%);
	}
}

@keyframes highlight {
	0% {
		background-position: 100%;
	}
}
</style>
