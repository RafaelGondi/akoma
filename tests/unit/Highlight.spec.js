import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import Highlight from '../../src/components/Highlight.vue';

const localVue = createLocalVue();

test('Component is mounted properly', () => {
	const wrapper = mount(Highlight, {
		localVue,
	});

	expect(wrapper).toMatchSnapshot();
});

