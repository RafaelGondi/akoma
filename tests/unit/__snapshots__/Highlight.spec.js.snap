// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `<span class="highlight__container  highlight__container--info" style="--duration: 1s; --delay: 0s;"></span>`;
